frontend/server/src/main/java/org/pytorch/serve/http/api/rest/InferenceRequestHandler.java

```diff
@@ -59,10 +59,13 @@ public void handleRequest(
             String[] segments)
             throws ModelException, DownloadArchiveException, WorkflowException,
                     WorkerInitializationException {
+        ConfigManager configManager = ConfigManager.getInstance();
         if (isInferenceReq(segments)) {
             if (endpointMap.getOrDefault(segments[1], null) != null) {
+                configManager.checkTokenAuthorization(req, false);
                 handleCustomEndpoint(ctx, req, segments, decoder);
             } else {
+                configManager.checkTokenAuthorization(req, true);
```

Comment:
the implementation token authentication should not be spreaded in each endpoint. It should be handled by InvalidRequestHandler.